### ISSUE TITLE
RSpec/NamedSubject: false positive when raising error

### ISSUE THREAD

**Issue by cabello** [rubocop-hq/rubocop#6825]
## Expected behavior

To not trigger when the following construction is used:

```ruby
it { expect { subject }.to raise_error(ArgumentError, 'problem') }
```

## Actual behavior

Suggests to name subject:

```
RSpec/NamedSubject: Name your test subject if you need to reference it explicitly.
  it { expect { subject }.to raise_error(ArgumentError, 'problem') }
```

## Steps to reproduce the problem

Enable `RSpec/NamedSubject` in your `.rubocop.yml`:

```yaml
RSpec/NamedSubject:
  Enabled: true
```

Add a spec that uses subject in a block to check for exception:

```ruby
RSpec.describe Object do
  subject { raise ArgumentError, 'problem' }

  it { expect { subject }.to raise_error(ArgumentError, 'problem') }
end
```

## RuboCop version

```bash
$ bundle exec rubocop -V
0.65.0 (using Parser 2.6.0.0, running on ruby 2.4.5 x86_64-darwin18)
```

**Comment by rrosenblum** [comment]
The RSpec cops live in the [`rubocop-rspec`](https://github.com/rubocop-hq/rubocop-rspec) project. Please open up an issue in that project.
